cmd/nodepool/core/create.go

```diff
@@ -0,0 +1,116 @@
+package core
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+
+	hyperv1 "github.com/openshift/hypershift/api/v1alpha1"
+	"github.com/openshift/hypershift/cmd/util"
+	hyperapi "github.com/openshift/hypershift/support/api"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	crclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type CreateNodePoolOptions struct {
+	Name         string
+	Namespace    string
+	ClusterName  string
+	NodeCount    int32
+	ReleaseImage string
+	Render       bool
+}
+
+// ApplyPlatformSpecifics can be used to create platform specific values as well as enriching the fixure with additional values
+type CreateNodePoolPlatformOptions interface {
```

Comment:
naming seems redundant given we are in nodepool/core/create, may be just
`
type PlatformOptions interface {
 Update()
 type()
}
`
wdyt?

Reply:
Renaming `CreateNodePoolPlatformOptions` to `PlatformOptions`  and `platformType` to `Type` sound good to me, but I'm not sure about renaming `UpdatePlatformParams` to `Update`, because it can be interpreted as we this function is updating the `PlatformOptions` itself
What do you think about changing it to `UpdateNodePool`?
```
type PlatformOptions interface {                                                                                                                       
        // UpdateNodePool is used to update the platform specific values in the NodePool                                                               
        UpdateNodePool(ctx context.Context, nodePool *hyperv1.NodePool, hcluster *hyperv1.HostedCluster, client crclient.Client) error
        // Type returns the platform type                                                                                                              
        Type() hyperv1.PlatformType                                                                                                                    
}